src/components/Editor.vue

```diff
@@ -642,6 +644,35 @@ export default {
 			this.emit('delete-image-node', imageUrl)
 		},
 
+		onYjsUpdate(update, _origin, _doc, _tr) {
+			// Update cursor labels for clients from awareness message
+			if (getMessageType(update) === yjsMessageTypes.awareness) {
+				if (!this.$editor) {
+					return
+				}
+
+				const clientIds = getAwarenessMessageClientIds(update)
+				const updateUsers = this.$editor.storage.collaborationCursor.users
+					.filter(u => clientIds.includes(u.clientId))
+				for (const user of updateUsers) {
+					this.updateCursorLabel(user)
+				}
+			}
+		},
+
+		updateCursorLabel: debounce(function(user) {
+			// Remove CSS class that fades out and hides the cursor label and bring it back after five seconds.
+			// Limit this to once every second to prevent it from happening with each awareness update.
+			const cursorLabelEls = document.getElementsByClassName(`collaboration-cursor__label__${user.name}`)
+			for (const el of cursorLabelEls) {
+				el.classList.remove('collaboration-cursor__label_fadeout')
+				clearTimeout(this.$cursorLabelTimeouts[user.clientId])
+				this.$cursorLabelTimeouts[user.clientId] = setTimeout(() => {
+					el.classList.add('collaboration-cursor__label_fadeout')
```

Comment:
Could there be cases where the element no longer exists here after the timeout?

Reply:
Good point. It's possible (because the session got closed). I'll check for the element and only apply the changes in case it exists.

Reply:
Well, given that I already get the list of elements with `getElementsByClassName` and iterate over them, the code should be fine as is, no?